components/accelerator/nvidia/temperature/component.go

```diff
@@ -143,8 +149,20 @@ func (c *component) CheckOnce() {
 		d.healthy = false
 		d.reason = fmt.Sprintf("exceeded HBM temperature thresholds: %s", strings.Join(tempThresholdExceeded, ", "))
 	}
+
+	return d
 }
 
+func CollectData(nvmlInstance nvidianvml.InstanceV2) (components.Data, error) {
+	d := collectData(nvmlInstance, nvidianvml.GetTemperature)
```

Comment:
This is what confuses me.. We pass a nvmlInstance, and then pass its method into the function.

Reply:
We need common function that can mock the get temp function, for the unit test coverage. In the future, I will clean up the nvml instance interface, so we don't need to pass this along.